### PR TITLE
added consts and minor auto refactor in upnp.cpp

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: include/libtorrent/upnp.hpp [hunk 1]
```diff
@@ -184,7 +184,7 @@ struct TORRENT_EXTRA_EXPORT upnp final
 	void discover_device();
 	void close();
 
-	// This is only available for UPnP routers. If the model is advertized by
+	// This is only available for UPnP routers. If the model is advertised by
```
Comment (arvidn) [inline include/libtorrent/upnp.hpp]: I added hunspell to RC_1_1 :) I will soon merged it into master

File: src/upnp.cpp [hunk 2]
```diff
@@ -1079,8 +1077,7 @@ void upnp::disable(error_code const& ec)
 	m_disabled = true;
 
 	// kill all mappings
-	for (std::vector<global_mapping_t>::iterator i = m_mappings.begin()
-		, end(m_mappings.end()); i != end; ++i)
+	for (auto i = m_mappings.begin(), end(m_mappings.end()); i != end; ++i)
```
Comment (arvidn) [inline src/upnp.cpp]: this could probably even be a range-for loop

**Comment by arvidn** [comment]
lgtm

File: src/upnp.cpp (hunk 2 above)
Comment (aldenml) [inline src/upnp.cpp]: this is a few lines below `int(i - m_mappings.begin()`

**Pull request merged**